app/src/main/java/org/simple/clinic/activity/TheActivity.kt

```diff
@@ -97,17 +102,22 @@ class TheActivity : AppCompatActivity() {
   private fun initialScreenKey(): FullScreenKey {
     val localUser = userSession.loggedInUser().blockingFirst().toNullable()
 
-    // TODO: Figure out how to handle the new PIN reset status
     val canMoveToHomeScreen = when (localUser?.loggedInStatus) {
       NOT_LOGGED_IN, RESETTING_PIN -> false
-      LOGGED_IN, OTP_REQUESTED, RESET_PIN_REQUESTED  -> true
+      LOGGED_IN, OTP_REQUESTED, RESET_PIN_REQUESTED -> true
       null -> false
     }
 
     return when {
       canMoveToHomeScreen -> HomeScreen.KEY
       hasUserCompletedOnboarding.get().not() -> OnboardingScreen.KEY
-      else -> RegistrationPhoneScreen.KEY
+      else -> {
+        return if (localUser?.loggedInStatus == RESETTING_PIN) {
+          ForgotPinCreateNewPinScreenKey()
+        } else {
+          RegistrationPhoneScreen.KEY
+        }
+      }
```

Comment:
This logic is slowly becoming complex. Although we cannot handle this through the controller because we need the initial screen key immediately, we could add a function to the controller for this.

```kotlin
private fun initialScreenKey() {
  return controller.initialScreenKey()
}
```

This way, the logic can also be tested. What do you think?

Reply:
I think it's a good idea. Will do this.